plugin/src/main/java/org/opensearch/ml/action/connector/SearchConnectorTransportAction.java

```diff
@@ -78,12 +81,38 @@ private void search(SearchRequest request, ActionListener<SearchResponse> action
                 excludes.toArray(new String[0])
             );
             request.source().fetchSource(rebuiltFetchSourceContext);
+
+            ActionListener<SearchResponse> doubleWrappedListener = ActionListener.wrap(wrappedListener::onResponse, e -> {
+                if (e instanceof IndexNotFoundException) {
+                    log
+                        .debug(
+                            "Connectors index not created yet, therefore we will swallow the exception and return an empty search result"
+                        );
+                    final InternalSearchResponse internalSearchResponse = InternalSearchResponse.empty();
+                    final SearchResponse emptySearchResponse = new SearchResponse(
+                        internalSearchResponse,
+                        null,
+                        0,
+                        0,
+                        0,
+                        0,
+                        null,
+                        new ShardSearchFailure[] {},
+                        SearchResponse.Clusters.EMPTY,
+                        null
+                    );
+                    wrappedListener.onResponse(emptySearchResponse);
```

Comment:
return empty result with status 200 may confusing user, better to keep rest status as 404 with customized error message.

Reply:
> return empty result with status 200 may confusing user, better to keep rest status as 404 with customized error message.

Why should user get or expect to get 404 if all they did was to ask which connectors are available?
The return of empty result is inline with the expected behavior described in the issue https://github.com/opensearch-project/ml-commons/issues/1787